File: src/_1ProgrammingBasicsWithJavaMay2023/_6NestedLoops/_2Exercise/_06CinemaTickets.java
package _1ProgrammingBasicsWithJavaMay2023._6NestedLoops._2Exercise;

import java.util.Scanner;

public class _06CinemaTickets {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String command = scanner.nextLine();
        int allTicketsSold = 0;
        int studentTickets = 0;
        int standardTickets = 0;
        int kidTickets = 0;
        while (!command.equals("Finish")) {
            String movieName = command;
            int seatsFree = Integer.parseInt(scanner.nextLine());
            int soldTicketsCounter = 0;

            String ticketsCommand = scanner.nextLine();
            while (!ticketsCommand.equals("End")) {
                String ticketType = ticketsCommand;
                soldTicketsCounter++;
                allTicketsSold++;
                switch (ticketType) {
                    case "student":
                        studentTickets++;
                        break;
                    case "standard":
                        standardTickets++;
                        break;
                    case "kid":
                        kidTickets++;
                        break;
                }
                if (soldTicketsCounter == seatsFree) {
                    break;
                }

                ticketsCommand = scanner.nextLine();
            }
            System.out.printf("%s - %.2f%% full.%n", movieName, 1.0 * soldTicketsCounter / seatsFree * 100);

            command = scanner.nextLine();
        }
        System.out.printf("Total tickets: %d%n", allTicketsSold);
        System.out.printf("%.2f%% student tickets.%n", 1.0 * studentTickets / allTicketsSold * 100);
        System.out.printf("%.2f%% standard tickets.%n", 1.0 * standardTickets / allTicketsSold * 100);
        System.out.printf("%.2f%% kids tickets.%n", 1.0 * kidTickets / allTicketsSold * 100);
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_6NestedLoops/_2Exercise/_06CinemaTickets.js
function cinemaTickets(input) {
    let command = input[0];

    let studentsTicket = 0;
    let standardTicket = 0;
    let kidTicket = 0;

    let index = 1;

    while (command !== "Finish") {
        const movieName = command;
        const numberOfSeats = input[index++];

        let soldTicketsForThisMovie = 0;

        while (soldTicketsForThisMovie < numberOfSeats) {
            const ticketsType = input[index++];

            if (ticketsType === "End") {
                break;
            }

            soldTicketsForThisMovie++;

            if (ticketsType === "student") {
                studentsTicket++;
            } else if (ticketsType === "standard") {
                standardTicket++;
            } else if (ticketsType === "kid") {
                kidTicket++;
            }
        }

        const percentSoldTickets = soldTicketsForThisMovie / numberOfSeats * 100;

        console.log(`${movieName} - ${percentSoldTickets.toFixed(2)}% full.`)

        command = input[index++];
    }

    const totalTickets = studentsTicket + standardTicket + kidTicket;
    console.log(`Total tickets: ${totalTickets}`);

    const percentStudentTickets = studentsTicket / totalTickets * 100;
    const percentStandardTickets = standardTicket / totalTickets * 100;
    const percentKidTickets = kidTicket / totalTickets * 100;

    console.log(`${percentStudentTickets.toFixed(2)}% student tickets.`);
    console.log(`${percentStandardTickets.toFixed(2)}% standard tickets.`);
    console.log(`${percentKidTickets.toFixed(2)}% kids tickets.`);
}